### DISCUSSION TITLE
Support 4-dimensional local accessor ?

### DISCUSSION THREAD

**Discussion by zjin-lcf** [intel/llvm#5238] [General]
Could a local accessor support dimensions larger than 3 ?  Thanks.

accessor<double, 4, sycl_read_write, access::target::local> mul_4d({4, batch, 1, 2}, cgh);
                                                                     
```
error: static_assert failed due to requirement '4 >= 1 && 4 <= 3' "range can only be 1, 2, or 3 dimensional."
  static_assert(dimensions >= 1 && dimensions <= 3,
  ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
sycl/CL/sycl/detail/accessor_impl.hpp:67:15: note: in instantiation of template class 'sycl::range<4>' requested here
  range<Dims> AccessRange;
              ^
sycl/CL/sycl/accessor.hpp:1865:48: note: in instantiation of template class 'sycl::detail::LocalAccessorBaseDevice<4>' requested here
  detail::LocalAccessorBaseDevice<AdjustedDim> impl;
```

**Comment by AlexeySachkov** [comment]
According to SYCL 2020 spec, section [4.7.6.6. Accessing elements of an accessor](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_accessing_elements_of_an_accessor):

> Accessors of type accessor, host_accessor, and local_accessor can have zero, one, two, or three Dimensions.

So, in scope of a core functionality, this can't be implemented at the moment, requested functionality should be formulated as an extension first. Tagging @jbrodman and @gmlueck for awareness.

AFAIK, @keryell has a prototype for this in triSYCL/triSYCL#313

**Reply by zjin-lcf** [reply]
Thanks for your answer.
